rust/src/compile.rs

```diff
@@ -0,0 +1,94 @@
+use crate::bs_safe::{BridgeStanError, Result};
+use log::info;
+use path_absolutize::Absolutize;
+use std::path::{Path, PathBuf};
+
+/// Compile a Stan Model given the path to BridgeStan and to a stan_file
+pub fn compile_model<P>(
+    bs_path: P,
+    stan_file: P,
+    stanc_args: Vec<&str>,
+    make_args: Vec<&str>,
+) -> Result<PathBuf>
+where
+    P: AsRef<Path>,
```

Comment:
Question as much for @aseyboldt as @randommm: The `AsRef<Path>` generic is the canonical example I have seen for the [Non-Generic Inner Function](https://www.possiblerust.com/pattern/non-generic-inner-functions) pattern: do we care about doing something like that here? I have no sense of if this is useful outside things like stdlib functions you expected to get called in many different ways

Reply:
fair point, idk, it was just a convenience to the user, but i think it's not worth the need for monomorphisations and such, so i've removed it.